libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java

```diff
@@ -833,11 +833,12 @@ public void run() {
     }
 
     @Override
-    public void onMediaUploadFailed(final String mediaId) {
+    public void onMediaUploadFailed(final String mediaId, final String errorMessage) {
         mWebView.post(new Runnable() {
             @Override
             public void run() {
-                mWebView.execJavaScriptFromString("ZSSEditor.markImageUploadFailed(" + mediaId + ");");
+                mWebView.execJavaScriptFromString("ZSSEditor.markImageUploadFailed(" + mediaId + ", '"
+                        + errorMessage + "');");
```

Comment:
Here `errorMessage` needs to have single quotes escaped. The permission error ('You don't have permission to upload media to the site'), for example, causes an error. `errorMessage.replaceAll("'", "\\\\'")` should do it.

We should probably do this everywhere we're sending string literals from Android, since a translated String may contain a single quote even if the default one doesn't.